packages/store/src/lib/actions.ts

```diff
@@ -0,0 +1,11 @@
+export enum EntityActions {
+  Set = 'Set',
+  Add = 'Add',
+  Update = 'Update',
+  Remove = 'Remove',
+}
+
+export interface Action<IDType> {
```

Comment:
The base action should have only a `type`.

Reply:
We can add a payload key:

export interface Action<T extends string, Payload> {
  type: T,
  payload: Payload
}

Reply:
I'd still need to use the `Action` type in the store class, I'm not sure I can think of a way where store and entities are decoupled.

```ts
actions$: Observable<Action<EntityActions, EntityIdType>> =
    this.context.actions.asObservable();
```

The same problem with inferring the EntityIdType, the store has to have a bit of knowledge about the entities to figure out the correct type.

I'm not that great with typescript so any help is appreciated!

Reply:
We can probably leave it as `Observable<Action<any, any>>` and let the `ofType` function to narrow down the type